### ISSUE TITLE
bug: CloudFormation stack creation always CREATE_FAILED

### ISSUE THREAD

**Issue by wantdrink** [localstack/localstack#4528]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Hi there,

I'm trying to create a stack but always get CREATE_FAILED result.
I deleted most of the resource in stack template to make it simple:

```
Metadata: 
  License: Apache-2.0
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  EC2Instance:
    Type: AWS::EC2::Instance
    Properties:
      InstanceType: "t3.small"
      ImageId: "ami-c6f321bf"

```

The stack result:
```
{
    "StackSummaries": [
        {
            "StackId": "arn:aws:cloudformation:us-east-1:000000000000:stack/EIP_With_Association.yaml1/11e09c7e",
            "StackName": "EIP_With_Association.yaml1",
            "CreationTime": "2021-09-02T06:32:39.457Z",
            "StackStatus": "CREATE_FAILED",
            "StackStatusReason": "Deployment failed"
        }
    ]
}
```

The services in docker-compose.yml is :
```
    environment:
      - SERVICES=serverless,sqs,sns,cloudformation,ec2,s3,ebs,lambda,sts,iam,apigateway,cloudwatch

```
Image info:

> image localstack/localstack            latest                    8181ed740500        8 days ago

Thanks a lot.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker-compose up -d

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

create stack with boto3

### Environment

```markdown
- OS: CentOS 7.7
- LocalStack: latest (image localstack/localstack            latest                    8181ed740500        8 days ago)
```

**Comment by whummer** [comment]
Thanks for reporting @wantdrink . Can you please share your `docker-compose.yml` config file, as well as any error messages you're seeing the container logs (ideally with `DEBUG=1` enabled in the LocalStack container environment). Thanks!

**Comment by wantdrink** [comment]
Thank you @whummer for your reply.
Here is my docker-compose.yml

```
services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack:latest
    network_mode: bridge
    ports:
      - "127.0.0.1:53:53"
      - "127.0.0.1:53:53/udp"
      - "127.0.0.1:443:443"
      - "127.0.0.1:4566:4566"
      - "127.0.0.1:4571:4571"
    environment:
      - SERVICES=serverless,sqs,sns,cloudformation,ec2,s3,ebs,lambda,sts,iam,apigateway,cloudwatch
      - DEBUG=1
      - DATA_DIR=/localstack
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - LOCALSTACK_API_KEY=${LOCALSTACK_API_KEY- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOST_TMP_FOLDER=/tmp
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"


```
And here is the error logs:

```
localstack_main | 2021-09-06T09:46:30:DEBUG:localstack.services.cloudformation.cloudformation_api: Creating stack "stack2" with 1 resources ...
localstack_main | 2021-09-06 09:46:30,253:API: 127.0.0.1 - - [06/Sep/2021 09:46:30] "POST / HTTP/1.1" 200 -
localstack_main | 2021-09-06T09:46:30:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "EC2Instance" (1/2) type "AWS::EC2::Instance" in loop iteration 1
localstack_main | 2021-09-06T09:46:30:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "EC2::Instance" id "EC2Instance"
localstack_main | 2021-09-06T09:46:30:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "EC2::Instance" in region us-east-1: create_instances {'InstanceType': 't3.small', 'ImageId': 'ami-03cf127a', 'MinCount': 1, 'MaxCount': 1}
localstack_main | 2021-09-06 09:46:30,306:API: 127.0.0.1 - - [06/Sep/2021 09:46:30] "POST / HTTP/1.1" 200 -
localstack_main | 2021-09-06 09:46:30,348:API: 127.0.0.1 - - [06/Sep/2021 09:46:30] "POST / HTTP/1.1" 200 -
localstack_main | 2021-09-06 09:46:30,397:API: 127.0.0.1 - - [06/Sep/2021 09:46:30] "POST / HTTP/1.1" 200 -
localstack_main | 2021-09-06 09:46:30,466:API: 127.0.0.1 - - [06/Sep/2021 09:46:30] "POST / HTTP/1.1" 200 -
localstack_main | 2021-09-06T09:46:30:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Modify" for resource "EC2Instance" (1/1) type "AWS::EC2::Instance" in loop iteration 1
localstack_main | 2021-09-06T09:46:30:INFO:localstack.utils.cloudformation.template_deployer: Updating resource EC2Instance of type EC2::Instance
localstack_main | 2021-09-06T09:46:30:DEBUG:localstack.utils.cloudformation.template_deployer: Error applying changes for CloudFormation stack "stack2": Parameter validation failed:
localstack_main | Invalid type for parameter Groups, value: None, type: <class 'NoneType'>, valid types: <class 'list'>, <class 'tuple'> Traceback (most recent call last):
localstack_main |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 2138, in _run
localstack_main |     self.do_apply_changes_in_loop(changes, stack, stack_name)
localstack_main |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 2209, in do_apply_changes_in_loop
localstack_main |     self.apply_change(change, stack, new_resources, stack_name=stack_name)
localstack_main |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 2277, in apply_change
localstack_main |     result = update_resource(resource_id, new_resources, stack_name)
localstack_main |   File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 997, in update_resource
localstack_main |     return instance.update_resource(resource, stack_name=stack_name, resources=resources)
localstack_main |   File "/opt/code/localstack/localstack/services/cloudformation/service_models.py", line 2161, in update_resource
localstack_main |     InstanceType={"Value": props["InstanceType"]},
localstack_main |   File "/opt/code/localstack/.venv/lib/python3.7/site-packages/botocore/client.py", line 386, in _api_call
localstack_main |     return self._make_api_call(operation_name, kwargs)
localstack_main |   File "/opt/code/localstack/.venv/lib/python3.7/site-packages/botocore/client.py", line 678, in _make_api_call
localstack_main |     api_params, operation_model, context=request_context)
localstack_main |   File "/opt/code/localstack/.venv/lib/python3.7/site-packages/botocore/client.py", line 726, in _convert_to_request_dict
localstack_main |     api_params, operation_model)
localstack_main |   File "/opt/code/localstack/.venv/lib/python3.7/site-packages/botocore/validate.py", line 319, in serialize_to_request
localstack_main |     raise ParamValidationError(report=report.generate_report())
localstack_main | botocore.exceptions.ParamValidationError: Parameter validation failed:
localstack_main | Invalid type for parameter Groups, value: None, type: <class 'NoneType'>, valid types: <class 'list'>, <class 'tuple'>
localstack_main | 
localstack_main | 2021-09-06 09:47:00,378:API: 127.0.0.1 - - [06/Sep/2021 09:47:00] "POST / HTTP/1.1" 200 -

```
Looks like there must be a parameter groups in the template?

Thanks.

**Comment by DVader666** [comment]
I'm getting a similar issue while using the LocalStack build for today:

```
LocalStack version: 0.12.18
LocalStack Docker container id: aed209e13423
LocalStack build date: 2021-10-06
LocalStack build git hash: 487e80ce
```
Whereas the same project was deploying OK five days ago. I mean, other team members are currently using LocalStack image from five days ago and they don't have the same issue (unfortunately, I decided to pull latest localstack/localstack image today, along with another guy from the team). The other guys have the same LocalStack version but a previous build:

```
LocalStack version: 0.12.18
LocalStack Docker container id: 65df8c5709ed
LocalStack build date: 2021-10-01
LocalStack build git hash: 69dd7992
```
We are deploying our project via Serverless Framework, and as I said, everything is working OK for the guys in the Team using the LocalStack image from 2021-10-01 with hash 69dd7992.

The issue we get in LocalStack log (DEBUG=1) is as follows:

```
2021-10-06T22:21:38:DEBUG:localstack.utils.cloudformation.template_deployer: Error applying changes for CloudFormation stack "local-whatever-stack": 'properties' Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1485, in _run
    self.do_apply_changes_in_loop(changes, stack, stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1518, in do_apply_changes_in_loop
    existing_resources=new_resources,
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1129, in add_default_resource_props
    resource_class.add_defaults(resource, stack_name)
  File "/opt/code/localstack/localstack/services/cloudformation/models/apigateway.py", line 63, in add_defaults
    role_name = resource["properties"].get("Name")
KeyError: 'properties'
```

**Comment by whummer** [comment]
Thanks for reporting! 

@DVader666 this issue should be fixed in the meantime - can you please pull the `latest` Docker image and give it another try?

@wantdrink - we'll look into this issue shortly.. 👍 Thanks

**Comment by DVader666** [comment]
Verified. All good. Thank you.

**Comment by whummer** [comment]
Thanks again for the update @wantdrink . This issue should be fixed via #5117 - the `latest` Docker image is currently being built in our CI pipeline and will become available in ~45mins. Please let us know if the problem persists.. Thanks!
